src/pkg/cli/compose/validation.go

```diff
@@ -285,6 +291,11 @@ func ValidateProject(project *compose.Project) error {
 	for k := range project.Extensions {
 		term.Warnf("unsupported compose extension: %q", k)
 	}
+
+	if numOfManagedServices > 0 && numOfManagedServices == len(project.Services) {
+		return ErrOnlyManagedServicesDefined
```

Comment:
This isn't an error so we shouldn't user errors for this. Frankly I'd just add code to `commands.go`, iterate the services and check them there, instead of this diff which is way too many lines for what we're trying to do.